module-apps/application-phonebook/widgets/InputLinesWithLabelWidget.cpp

```diff
@@ -68,20 +68,36 @@ namespace gui
                 inputText->setCursorStartPosition(CursorStartPosition::DocumentEnd);
                 inputText->setUnderlineThickness(style::window::default_border_focus_w);
                 inputText->setFont(style::window::font::mediumbold);
+
+                if (!inputText->isEmpty() || Clipboard::getInstance().gotData()) {
+                    this->navBarTemporaryMode(utils::translate(style::strings::common::options), false);
+                }
             }
             else {
                 inputText->setCursorStartPosition(CursorStartPosition::DocumentBegin);
                 inputText->setUnderlineThickness(style::window::default_border_rect_no_focus);
                 inputText->setFont(style::window::font::medium);
+
+                this->navBarRestoreFromTemporaryMode();
             }
             return true;
         };
 
         inputCallback = [&](Item &item, const InputEvent &event) {
             auto result = inputText->onInput(event);
-            if (checkTextContent != nullptr) {
-                checkTextContent();
+
+            if (!event.isShortRelease(gui::KeyCode::KEY_AST) &&
+                (!inputText->isEmpty() || Clipboard::getInstance().gotData())) {
+                this->navBarTemporaryMode(utils::translate(style::strings::common::options), false);
+            }
+
+            if (event.isShortRelease(gui::KeyCode::KEY_LF) &&
+                (!inputText->isEmpty() || Clipboard::getInstance().gotData())) {
```

Comment:
possibly clipboard should be used with some dependency injection

Reply:
Hmm need more information :) or that is just general remark to refactor Clipboard one day ?